src/main/java/org/kohsuke/github/GHCheckRunBuilder.java

```diff
@@ -0,0 +1,370 @@
+import java.util.Collections;
+import java.util.Date;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * Drafts a check run.
+ *
+ * @see GHCheckRun
+ * @see GHRepository#createCheckRun
+ * @see <a href="https://developer.github.com/v3/checks/runs/#create-a-check-run">documentation</a>
+ */
+@SuppressFBWarnings(value = "URF_UNREAD_FIELD", justification = "Jackson serializes these even without a getter")
+@Preview
+@Deprecated
+public final class GHCheckRunBuilder {
+
+    private final GHRepository repo;
+    private final Requester requester;
+    Output output;
+    List<Action> actions;
+
+    GHCheckRunBuilder(GHRepository repo, String name, String headSHA) {
+        this.repo = repo;
+        requester = repo.root.createRequest()
+                .withPreview(Previews.ANTIOPE)
+                .method("POST")
+                .with("name", name)
+                .with("head_sha", headSHA)
+                .withUrlPath(repo.getApiTailUrl("check-runs"));
+    }
+
+    public @NonNull GHCheckRunBuilder withDetailsURL(@CheckForNull String detailsURL) {
+        requester.with("details_url", detailsURL);
+        return this;
+    }
+
+    public @NonNull GHCheckRunBuilder withExternalID(@CheckForNull String externalID) {
+        requester.with("external_id", externalID);
+        return this;
+    }
+
+    public @NonNull GHCheckRunBuilder withStatus(@CheckForNull GHCheckRun.Status status) {
+        if (status != null) {
+            // Do *not* use the overload taking Enum, as that s/_/-/g which would be wrong here.
+            requester.with("status", status.toString().toLowerCase(Locale.ROOT));
+        }
+        return this;
+    }
+
+    public @NonNull GHCheckRunBuilder withConclusion(@CheckForNull GHCheckRun.Conclusion conclusion) {
+        if (conclusion != null) {
+            requester.with("conclusion", conclusion.toString().toLowerCase(Locale.ROOT));
+        }
+        return this;
+    }
+
+    public @NonNull GHCheckRunBuilder withStartedAt(@CheckForNull Date startedAt) {
+        if (startedAt != null) {
+            requester.with("started_at", GitHubClient.printDate(startedAt));
+        }
+        return this;
+    }
+
+    public @NonNull GHCheckRunBuilder withCompletedAt(@CheckForNull Date completedAt) {
+        if (completedAt != null) {
+            requester.with("completed_at", GitHubClient.printDate(completedAt));
+        }
+        return this;
+    }
+
+    /**
+     * Drafts the output section.
+     *
+     * @param title
+     *            as in GitHub documentation
+     * @param summary
+     *            as in GitHub documentation
+     * @return use {@link Output#done} to continue
+     */
+    public @NonNull Output withOutput(@NonNull String title, @NonNull String summary) {
+        return new Output(this, title, summary);
+    }
+
+    /**
+     * Drafts an action section.
+     *
+     * @param label
+     *            as in GitHub documentation
```

Comment:
What happens when the max size of this is exceeded?  Not mentioning that limit here will cause more pain for consumers of this library.

Reply:
I would rather not duplicate backend logic here. It is up to callers to read the GitHub documentation and supply appropriate parameters accordingly. The limits may change over time, remember, and since Checks is still in preview there could be _new_ restrictions later. The library should limit itself to collecting data and shipping it off.